plugin/HASH.ml

```diff
@@ -0,0 +1,26 @@
+open Constants
+open Constants.K
+
+let do_hash_str: string -> Cryptokit.hash -> Constants.K.kitem list =
+  fun str h ->
+    let bytes = Cryptokit.hash_string h str in
+    let buf = Buffer.create ((String.length bytes) * 2) in
+    String.iter (fun c -> Buffer.add_string buf (Printf.sprintf "%02x" (int_of_char c))) bytes;
+    [String (Buffer.contents buf)]
+
+let do_hash_bytes: bytes -> Cryptokit.hash -> Constants.K.kitem list =
+  fun bytes h ->
+    h#add_substring bytes 0 (Bytes.length bytes);
```

Comment:
I'm not familiar with this hash sign syntax but if your intent is just to have an overload that works over bytes I would recommend just calling `do_hash_str` on `Bytes.to_string bytes`.

Reply:
Oh wait I guess that wouldn't work since you want the output to be raw and not hex encoded. Still, can you explain what this syntax is so I can double check to make sure that it's supported in the right versions of OCAML?

Reply:
Its the first syntax I found when googling for "ocaml object call method". If there's another you prefer or are more familiar with, I'd be happy to change it.